src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -562,6 +583,8 @@
         this)
       (a/start [this]
         (run-lifecycle-fns app-context s/start "start" ordered-services)
+        (when-let [restart-file (get-in-config (a/get-service this :ConfigService) [:global :restart-file])]
+          (inc-restart-counter! restart-file))
```

Comment:
That sounds right (and, good catch!)... but, I'm a little confused because it appears that there is a test that is asserting that the counter gets incremented due to a restart, and that test is passing?